libs/template/config_test.go

```diff
@@ -194,3 +194,103 @@ func TestTemplateSchemaErrorsWithEmptyDescription(t *testing.T) {
 	_, err := newConfig(context.Background(), "./testdata/config-test-schema/invalid-test-schema.json")
 	assert.EqualError(t, err, "template property property-without-description is missing a description")
 }
+
+func TestPromptIsSkippedWhenEmpty(t *testing.T) {
+	c := config{
+		ctx:    context.Background(),
+		values: make(map[string]any),
+		schema: &jsonschema.Schema{
+			Properties: map[string]*jsonschema.Schema{
+				"always-skip": {
+					Type:    "string",
+					Default: "some-default-value",
+					Extension: jsonschema.Extension{
+						SkipPromptIf: &jsonschema.Schema{},
+					},
+				},
+			},
+		},
+	}
+
+	// We should always skip the prompt here. An empty JSON schema by definition
+	// matches all possible configurations.
+	assert.True(t, c.isSkipped(jsonschema.Property{
+		Name:   "always-skip",
+		Schema: c.schema.Properties["always-skip"],
+	}))
+	assert.Equal(t, "some-default-value", c.values["always-skip"])
+}
```

Comment:
What is the expected behavior if the value is already set?

The current implementation overwrites the value with the default, even if it is already set.

See comment line `// Config already has a value assigned`.

Reply:
Done. You were right. If the configuration is assigned now we will not evaluate the skip condition for it.